view/src/components/deal.js

```diff
@@ -0,0 +1,479 @@
+   * @param data A deal object
+   */
+  handleDelete(data) {
+    authMiddleWare(this.props.history);
+    const authToken = localStorage.getItem("AuthToken");
+    axios.defaults.headers.common = { Authorization: `${authToken}` };
+    let todoId = data.todo.todoId;
+    axios
+      .delete(`todo/${todoId}`) // NOT CONNECTED TO DEALS CRUD YET
+      .then(() => {
+        window.location.reload();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  /**
+   * Takes a deal object as an input and opens a dialog page to
+   * allow the user to update the attributes of the deal object
+   * @param data A deal object
+   */
+  handleEditClick(data) {
+    this.setState({
+      // Deal state
+      title: data.todo.title,
+      body: data.todo.body,
+      todoId: data.todo.todoId,
+      // Page state
+      buttonType: "Edit",
+      open: true,
+    });
+  }
+
+  /**
+   * Takes a deal object as an input and opens a popup with all the
+   * information about the deal (currently not being used -> will be
+   * updated to show augmented information)
+   * @param data A deal object
+   */
+  handleViewOpen(data) {
+    this.setState({
+      // Deal state
+      title: data.todo.title,
+      body: data.todo.body,
+      // Page state
+      viewOpen: true,
+    });
+  }
+
+  render() {
+    const DialogTitle = withStyles(styles)((props) => {
+      const { children, classes, onClose, ...other } = props;
+      return (
+        <MuiDialogTitle disableTypography className={classes.root} {...other}>
+          <Typography variant="h6">{children}</Typography>
+          {onClose ? (
+            <IconButton
+              aria-label="close"
+              className={classes.closeButton}
+              onClick={onClose}
+            >
+              <CloseIcon />
+            </IconButton>
+          ) : null}
+        </MuiDialogTitle>
+      );
+    });
+
+    const DialogContent = withStyles((theme) => ({
+      viewRoot: {
+        padding: theme.spacing(2),
+      },
+    }))(MuiDialogContent);
+
+    dayjs.extend(relativeTime);
+    const { classes } = this.props;
+    const { open, errors, viewOpen } = this.state;
+
+    /** Set states related to dialogue to generic value when opening */
+    const handleAddClick = () => {
+      this.setState({
+        // Deal states
+        todoId: "",
+        title: "",
+        body: "",
+        buttonType: "",
+        // Page state
+        open: true,
+      });
+    };
+
+    /**
+     * Either updates or submits a new deal object to the data base
+     * @param event The event being handled
+     */
+    const handleSubmit = (event) => {
+      event.preventDefault();
+      const userTodo = {
+        // Deal states
+        title: this.state.title,
+        body: this.state.body,
+      };
+      let options = {};
+      if (this.state.buttonType === "Edit") {
+        options = {
+          url: `/todo/${this.state.todoId}`, // NOT CONNECTED TO CRUD YET
```

Comment:
Same here, let's not shout in the comments and let's avoid todo if possible (goes for the entire file)